concent_api/api_testing_common.py

```diff
@@ -0,0 +1,187 @@
+    "ProtocolConstants",
+    [
+        "concent_messaging_time",
+        "subtask_verification_time",
+        "force_acceptance_time",
+        "token_expiration_time",
+    ]
+)
+
+
+def get_protocol_constants(cluster_url):
+    url                         = f"{cluster_url}/api/v1/protocol-constants/"
+    response                    = requests.get(url)
+    json                        = response.json()
+    concent_messaging_time      = json['concent_messaging_time']
+    subtask_verification_time   = json['subtask_verification_time']
+    force_acceptance_time       = json['force_acceptance_time']
+    token_expiration_time       = json['token_expiration_time']
+    constants                   = ProtocolConstants(
+            concent_messaging_time      = concent_messaging_time,
+            subtask_verification_time   = subtask_verification_time,
+            force_acceptance_time       = force_acceptance_time,
+            token_expiration_time       = token_expiration_time
+    )
+    return constants
+
+
+def print_protocol_constants(constants):
+    print("PROTOCOL_CONSTANTS: ")
+    print(f"concent_messaging_time = {constants.concent_messaging_time}")
+    print(f"subtask_verification_time = {constants.subtask_verification_time}")
+    print(f"force_acceptance_time = {constants.force_acceptance_time}")
+    print(f"token_expiration_time = {constants.token_expiration_time}\n")
+
+
+def print_golem_message(message, indent=4):
+    assert isinstance(message, Message)
+    HEADER_FIELDS  = ['timestamp', 'encrypted', 'sig']
+    PRIVATE_FIELDS = {'_payload', '_raw'}
+    assert 'type' not in message.__slots__
+    fields = ['type'] + HEADER_FIELDS + sorted(set(message.__slots__) - set(HEADER_FIELDS) - PRIVATE_FIELDS)
+    values = [
+        type(message).__name__ if field == 'type' else
+        getattr(message, field)
+        for field in fields
+    ]
+
+    for field, value in zip(fields, values):
+        if isinstance(value, Message):
+            print_golem_message(value, indent=indent + 4)
+        elif field == 'timestamp':
+            print('{}{:30} = {}  # UTC: {}'.format(' ' * indent, field, value, timestamp_to_isoformat(value)))
+        else:
+            print('{}{:30} = {}'.format(' ' * indent, field, value))
+
+
+def validate_response_status(actual_status_code, expected_status):
+    if expected_status is not None:
+        assert expected_status == actual_status_code, f"Expected:HTTP{expected_status}, actual:HTTP{actual_status_code}"
+
+
+def validate_response_message(encoded_message, expected_message, private_key, public_key,):
+    if expected_message is not None:
+        decoded_message = try_to_decode_golem_message(private_key, public_key, encoded_message)
+        assert isinstance(decoded_message, expected_message), f"Expected:{expected_message}, actual:{decoded_message}"
+
+
+def api_request(host, endpoint, private_key, public_key, data=None, headers=None, expected_status=None, expected_message=None):
```

Comment:
We need `expected_content_type` here. Or it might be assumed based on the data type. Right now it's not checked at all.

Reply:
done